src/semidiscretization/semidiscretization.jl

```diff
@@ -251,15 +251,6 @@ function jacobian_ad_forward(semi::AbstractSemidiscretization, t0, u0_ode)
     _jacobian_ad_forward(semi, t0, u0_ode, du_ode, config)
 end
 
-function _jacobian_ad_forward(semi, t0, u0_ode, du_ode, config)
-    new_semi = remake(semi, uEltype = eltype(config))
-    J = ForwardDiff.jacobian(du_ode, u0_ode, config) do du_ode, u_ode
-        Trixi.rhs!(du_ode, u_ode, new_semi, t0)
-    end
-
-    return J
-end
-
```

Comment:
It's quite confusing for me that most stuff stays here but only this method is moved to the hyperbolic semidiscretization. I think this method may also work for more general semidiscretizations that we have already. Thus, couldn't we just keep this here and add the new version for hyperbolic-parabolic equations?

Reply:
Resolved in [73d387a](https://github.com/trixi-framework/Trixi.jl/pull/1589/commits/73d387aa281295b5014feabfde1330adaff7b8be)